js/background.js

```diff
@@ -107,9 +107,11 @@ function requestRedirected (info) {
 //
 async function finishRequest (info) {
     await requestUpdated(info);
-
-    info.completed = true;
-    info.timeCompleted = new Date().getTime();
+    
+    if (info.requestId && requests[info.requestId]) {
+        requests[info.requestId].completed = true;
```

Comment:
Well, I think here it is.
I noticed that the uploads did not occur at all. I think this is due to asynchronous nature. But we can modify the request stack directly. Given that finishRequest called only by webRequest.onCompleted, and the uploads are done at intervals — we are golden

Reply:
Hmm, it was working for me without this change, for some reason. But this way is better.